subsys/net/lib/dns/mdns_responder.c

```diff
@@ -684,13 +750,74 @@ static int init_listener(void)
 	return !ok;
 }
 
-static int mdns_responder_init(void)
+static void mdns_net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+					uint32_t mgmt_event, struct net_if *iface)
+
 {
+	int ret;
+
+	ARG_UNUSED(ret); /* Potentially not used */
+
+	switch (mgmt_event) {
+	case NET_EVENT_IF_UP:
+
+#if defined(CONFIG_NET_IPV4)
+		iface_ipv4_setup(iface, &local_addr4.sin_addr);
+#endif /* CONFIG_NET_IPV4 */
+
+#if defined(CONFIG_NET_IPV6)
+		iface_ipv6_setup(iface, &local_addr6.sin6_addr);
+#endif /* CONFIG_NET_IPV6 */
+		break;
+
+#if defined(CONFIG_MDNS_RESPONDER_DNS_SD_ANNOUNCE)
+#if defined(CONFIG_NET_IPV4)
+	case NET_EVENT_IPV4_ADDR_ADD:
+		ret = send_sd_announce(ipv4, iface, AF_INET, NULL);
+		if (ret < 0) {
+			NET_DBG("Failed to announce IPv4 to iface %p", iface);
+		}
+		break;
+#endif /* CONFIG_NET_IPV4 */
+
+#if defined(CONFIG_NET_IPV6)
+	case NET_EVENT_IPV6_ADDR_ADD:
+	case NET_EVENT_IPV6_DAD_SUCCEED:
+		ret = send_sd_announce(ipv6, iface, AF_INET6, NULL);
```

Comment:
Wouldn't this cause the announcement to be sent twice if DAD is enabled (we'll get both events for a single IPv6 address then)?

Reply:
Right, I'll make it conditional.
The RFC has more requirements to aggregate responses before sending, but the current implementation doesn't have any timeout handling so I didn't want to introduce such a dependency.